### PR TITLE
Issue 3728 - Edit project name tab

### LINKED ISSUE #3728: V5 Project Settings (rename project)
### Description

In order to get a functional version of V5 up and running we must have a means to access the project settings. Since the project settings only consists of renaming the project this can be built from scratch rather than porter from V4

### Goals

- [ ] As an admin I want to see the name of the project in the editable text field when I load the settings
- [ ] As an admin when I save a new project name I expect the Project name on the page and the tab header to update to the new name
- [ ] As an admin I expect other users to see the updated name when they next refresh/load the project page or teamspace project list

### PR DESCRIPTION
**Pull request by Amantini1997**
This fixes #3728

#### Description
Created the modal to edit a project

### CODE REVIEW

**Comment by Amantini1997** [comment]
> @Amantini1997 did you do the modal here? [#3728 (comment)](https://github.com/3drepo/3drepo.io/issues/3728#issuecomment-1302145343)
> 
> Because it shouldn't be a modal, it should've been a tab tab like teamspace settings instead and @george-hadfield should've updated the mock up......
> 
> Ignore me if you did the right thing.

I assumed that mock up was up to date, so I used that one indeed

**Comment by carmenfan** [comment]
> > @Amantini1997 did you do the modal here? [#3728 (comment)](https://github.com/3drepo/3drepo.io/issues/3728#issuecomment-1302145343)
> > Because it shouldn't be a modal, it should've been a tab tab like teamspace settings instead and @george-hadfield should've updated the mock up......
> > Ignore me if you did the right thing.
> 
> @carmenfan I assumed that mock up was up to date, so I used that one indeed I will fix it to match the new mock up then

the link is up to date but the screenshots should've been removed.

Figma should be the source of truth, so don't just blindly follow what's on github.

**Comment by Amantini1997** [comment]
> > > @Amantini1997 did you do the modal here? [#3728 (comment)](https://github.com/3drepo/3drepo.io/issues/3728#issuecomment-1302145343)
> > > Because it shouldn't be a modal, it should've been a tab tab like teamspace settings instead and @george-hadfield should've updated the mock up......
> > > Ignore me if you did the right thing.
> > 
> > 
> > @carmenfan I assumed that mock up was up to date, so I used that one indeed I will fix it to match the new mock up then
> 
> the link is up to date but the screenshots should've been removed.
> 
> Figma should be the source of truth for UI design, so don't just blindly follow what's on github.

Sure, I will try to pay more attention to future conflicts between images and links posted under an issue

**Comment by sanmont3drepo** [comment]
When I change tabs, project settings shows a required error.
reuired

File: frontend/test/projects/projects.sagas.spec.ts [hunk 1]
```diff
@@ -115,4 +117,34 @@ describe('Teamspaces: sagas', () => {
...
+				.dispatch(ProjectsActions.updateProject(teamspace, projectId, project, onSuccess, onError))
+				.put(ProjectsActions.updateProjectSuccess(teamspace, projectId, project))
+				.silentRun();
+
+			expect(onSuccess).toBeCalled();
+		});
+		it('should call updateProject endpoint with 404 and open alert modal', async () => {
+			mockServer
+				.patch(`/teamspaces/${teamspace}/projects/${projectId}`)
+				.reply(400)
```
Comment (The-Daniel) [inline frontend/test/projects/projects.sagas.spec.ts:141]: Doesn't really matter, but this should be 404 to match with the `it()`

**Review by The-Daniel — APPROVED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Review by The-Daniel — APPROVED** [review]

**Pull request merged**